cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedWorkflowProgramRunner.java

```diff
@@ -174,9 +178,19 @@ protected ProgramController launch(Program program, ProgramOptions options,
     return createProgramController(controller, program.getId(), runId);
   }
 
-  private static YarnConfiguration createConfiguration(YarnConfiguration hConf) {
+  private static YarnConfiguration createConfiguration(YarnConfiguration hConf, CConfiguration cConf,
+                                                       TokenSecureStoreRenewer secureStoreRenewer) {
     YarnConfiguration configuration = new YarnConfiguration(hConf);
     configuration.setBoolean(HCONF_ATTR_CLUSTER_MODE, true);
+    configuration.set("hive.metastore.token.signature", HiveAuthFactory.HS2_CLIENT_TOKEN);
+    if (SecurityUtil.isKerberosEnabled(cConf)) {
```

Comment:
This is porting the fix added in https://github.com/caskdata/cdap/commit/30d28e9f3ce31b0b818712659c2a126a9086c354#diff-e715a8969f2a3f1cc4c42dd5ff219b39R102